sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ProgressReporter.scala

```diff
@@ -345,3 +353,35 @@ trait ProgressReporter extends Logging {
     currentStatus = currentStatus.copy(message = message)
   }
 }
+
+/**
+ * A summary information of the streaming query.
+ */
+class QuerySummary {
```

Comment:
Looks like this class is always referenced with `QuerySummary.TOTAL_INPUT_RECORDS` (I meant single value), instead of looping and accessing available values. That makes me feel that it should be much simpler to just expose methods to get/update "total input records". At least for now, this seems overkill, as callers are not flexible to handle multiple values for summary.

Reply:
`QuerySummary` has been removed and all these UI data will be tracked in StreamingQueryListener. Mark this resolved.